docs/usage/automerge-configuration.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Automerge configuration and troubleshooting
+description: Learn all about Renovate's automerge functionality here
+---
+
+# Introduction
+
+Automerging is a Renovate feature that can enable you to fully automate upgrading of certain dependencies.
+When enabled, it means Renovate will attempt to merge the proposed update once it has passed tests.
+
+As a general guide, it is recommended that you enable automerge for any types of dependency updates where you would just click Merge anyway.
+If there are updates for which you prefer to manually review release notes or code before merging, even when they pass tests, then do not enable automerge for those.
+
+Automerge works particularly well for `devDependencies` as well as for production `dependencies` in projects which have great test coverage.
+
+For example, if you have `jest` or `mocha` as a dependency, and it has an upgrade with passing tests... automerge them!
+If you have a linter like `eslint` or `tslint` and its update passes... automerge them!
+If you have an API with 100% test coverage and `express` is updated... automerge it!
+
+## Configuration examples
+
+<!-- Internal comment
+
+I think the endgame here should be:
+
+- Each example use case is fully realised, and should be copy/pasteable and work.
+- The example should explain the reasoning/thinking behind each setting.
+- The example should have links to each setting, so that the docs can be consulted.
+
+-->
+
+### Automerge lock file maintenance
+
+The lowest risk type of update to automerge is probably `lockFileMaintenance`.
+When Renovate performs lock file maintenance, it means that it leaves the project dependency definitions unchanged, but refreshes the lock file completely so that latest versions according to package file constraints are installed.
+Here is an example of automerging lock file maintenance:
+
+```json
+{
+  "lockFileMaintenance": {
+    "enabled": true,
+    "automerge": true
+  }
+}
+```
+
+### Automerge lint tool updates
+
+Automerging lint tool updates can be a real time-saver.
+Sometimes an update to a lint tool or plugin definition can cause tests to fail, and that is usually deliberate/intentional because the lint authors have added a new validation that you need to adhere to.
```

Comment:
```suggestion
Sometimes an update to a lint tool or plugin definition can cause tests to fail, and that is usually deliberate/intentional because the lint authors have added a new rule that you need to adhere to.
```

Do we like `validation` or `rule` better here?